src/gsc_sql.c

```diff
@@ -0,0 +1,379 @@
+    }
+
+    if (mysql_query((MYSQL *)mysql, sql) == 0) {
+        Com_DPrintf("\nSuccessful query!\n");
+        MYSQL_RES *result = mysql_store_result((MYSQL *) mysql);
+
+        Scr_AddInt((int) result);
+    }
+    else
+        Scr_AddUndefined();
+}
+
+/* =================================================================
+* URL
+    http://dev.mysql.com/doc/refman/5.7/en/mysql-num-rows.html
+
+* Description
+    Returns the number of rows in the result set.
+
+* Return Value(s)
+    The number of rows in the result set.
+   ================================================================= */
+void gsc_mysql_rowcount() {
+    int result = Scr_GetInt(0);
+
+    if (Scr_GetNumParam() != 1) {
+        Scr_Error("Usage: gsc_mysql_rowcount(query result from gsc_mysql_query);\n");
+    }
+
+    int rowCount = mysql_num_rows((MYSQL_RES *) result);
+    Scr_AddInt( rowCount );
+}
+
+/* =================================================================
+* URL
+    http://dev.mysql.com/doc/refman/5.7/en/mysql-num-fields.html
+
+* Description
+    Returns the number of columns in a result set.
+
+* Return Value(s)
+    An unsigned integer representing the number of columns in a result set.
+   ================================================================= */
+void gsc_mysql_num_field() {
+    int result = Scr_GetInt(0);
+
+    if (Scr_GetNumParam() != 1) {
+        Scr_Error("Usage: gsc_mysql_num_field(query result from gsc_mysql_query);\n");
+    }
+
+    int fieldCount = mysql_num_fields((MYSQL_RES *) result);
+    Scr_AddInt( fieldCount );
+}
+
+/* =================================================================
+* URL
+    http://dev.mysql.com/doc/refman/5.7/en/mysql-fetch-row.html
+
+* Description
+    Retrieves the next row of a result set.
+
+* Return Value(s)
+    A MYSQL_ROW structure for the next row. NULL if there are no more rows to retrieve or if an error occurred.
+   ================================================================= */
+void gsc_mysql_fetch_rows() {
+    int results = Scr_GetInt(0);
+    MYSQL_RES* result = (MYSQL_RES*) results;
+
+    if (Scr_GetNumParam() != 1) {
+        Scr_Error("Usage: gsc_mysql_fetch_row(query result from gsc_mysql_query);\n");
+    }
+
+    if( mysql_num_rows(result) != 0 )
+    {
+        int num_fields = mysql_num_fields(result);
+        MYSQL_FIELD *field;
+        MYSQL_ROW row;
+        MYSQL_FIELD* fieldArray[num_fields];
+        int count = 0;
+        Scr_MakeArray();
+
+        /* Check to see if this is a single element row */
+        if (mysql_num_rows(result) == 1) {
+            while((field = mysql_fetch_field(result))) {
+                fieldArray[count] = field;
+                count++;
+            }
+            row = mysql_fetch_row(result);
+            Scr_MakeArray();
+            for (int i = 0; i < num_fields; i++) {
+                Scr_AddString(row[i]);
+                int tempId = Scr_AllocString(fieldArray[i]->name);
```

Comment:
Scr_AllocString can cause overflow because strings are not freed. Its better to allocate strings only once at initialization and save indexes returned by this function.